### PR TITLE
Build refactor

### PR DESCRIPTION
**Pull request by jamesbarnett91**
Migrate from Maven to Gradle, and build in Drone rather than Jenkins, to match out standard setup.

Bumped up the base image to Java 17, and made a few tweaks so JVM args can now be passed in from the env rather than having to override the entry point in the deployment.

No changes to app functionality. Briefly tested the image and all seems fine.

Jenkins previously tagged the image as <git tag>-<build number> via some convoluted methods e.g. `1.8-123`, or `1.8-auto-123` for the auto rebuilds.
We don't touch this service often and I don't think the version numbers and git tags really have much value, so I've simplified this to tag as as just <branch>-<build> number e.g. `master-123` or `master-auto-123`
This means the git tag is no longer needed, and we don't need to increment up the build number to match the Jenkins ones. 

We'll need to update the kube image policy to match the new pattern, I'll do that once merged.
This is currently set up to auto rebuild weekly using the digital team pattern of the `master-auto-` branch with incrementing build number. Given the low churn on this app I think that's fine but we could use the more complex eCase patten with the version number if preferred.

Also note that this repo is public so I've made sure the Drone access is set to private (by default it mirrors the repo visibility). I've also disabled builds on forks to prevent untrusted builds potentially trying to extract secrets. 
If I've forgotten anything else security wise please let me know

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		attributes 'Main-Class': 'uk.co.fivium.dmda.server.SMTPStart'
+	}
+}
+
+tasks.withType(JavaCompile) {
+	options.encoding = 'UTF-8'
+}
+
+test {
+	// hack to get https://github.com/stefanbirkner/system-lambda working with Java 11+
```
Comment (JayCGildea) [inline build.gradle:48]: It looks like the java version above is specified as 8 - why is this needed?

**Review by JayCGildea — APPROVED** [review]
LGTM - I can't think of any other issues with the secrets. I don't have any issue with the build pipeline given the once every 3 years we change this

File: build.gradle (hunk 1 above)
Comment (jamesbarnett91) [reply build.gradle:48]: The reference to 8 above is for the language level. I'm building and running on 17 now.
I can bump up the allowed language level to 17 to match as well, but figured since the existing code is all Java 8 we may as well keep 8 compatibility should we (or the unlikely event that anyone else who uses this) want to run on 8.

Next time someone does any work on this we can bump it up if they want to use 11+ features.

**Comment by benbasson** [comment]
Not fussy about the version number pattern - as long as we can move to automatic OS updates then I'm happy.

**Review by benbasson — APPROVED** [review]
